crates/ty_python_semantic/src/types.rs

```diff
@@ -4547,6 +4548,36 @@ impl<'db> Type<'db> {
             .to_instance(db)
             .expect("type should be convertible to instance type");
 
+        // Check if __new__ explicitly returns a different type than the instance
+        let new_return_type = new_call_outcome
+            .as_ref()
+            .and_then(|outcome| outcome.as_ref().ok())
+            .and_then(|bindings| {
+                let ret_type = bindings.return_type(db);
+
+                // Skip if return type is unknown/dynamic or None - likely unresolved annotations
+                if ret_type.is_unknown()
+                    || matches!(ret_type, Type::Dynamic(_))
+                    || ret_type.is_none(db)
```

Comment:
`__new__` returning `None` seems like an error, but I'm not sure why we should exclude it here. Unresolved annotations won't resolve to `None`.